### PR TITLE
Updated the Address class

### PR DESCRIPTION
**Pull request by popenta**
Updated the `Address` class to match the specs described [here](https://github.com/multiversx/mx-sdk-specs/blob/main/sdk-core/address.md).

### CODE REVIEW

File: multiversx_sdk_core/address.py [hunk 1]
```diff
@@ -1,94 +1,106 @@
...
 from multiversx_sdk_core.constants import DEFAULT_HRP, METACHAIN_ID
 from multiversx_sdk_core.errors import ErrBadAddress, ErrBadPubkeyLength
-from multiversx_sdk_core.interfaces import IAddress, INonce
 
 SC_HEX_PUBKEY_PREFIX = "0" * 16
 PUBKEY_LENGTH = 32
 PUBKEY_STRING_LENGTH = PUBKEY_LENGTH * 2  # hex-encoded
 BECH32_LENGTH = 62
 
+logger = logging.getLogger("Address")
```
Comment (ssd04) [inline multiversx_sdk_core/address.py:15]: in other file the logger name is with lower case, maybe keep the same format for name?

File: multiversx_sdk_core/address.py [hunk 2]
```diff
@@ -1,94 +1,106 @@
...
+        bytes_to_hash = deployer_pubkey + nonce_bytes
+        contract_pubkey = keccak.new(digest_bits=256).update(bytes_to_hash).digest()
+        contract_pubkey = bytes([0] * 8) + bytes([5, 0]) + contract_pubkey[10:30] + deployer_pubkey[30:]
+        return Address(contract_pubkey, deployer.get_hrp())
 
-    def pubkey_to_bech32(self, pubkey: bytes) -> str:
-        address = Address(pubkey, self.hrp)
-        return address.bech32()
+    def get_shard_of_address(self, address: IAddress) -> int:
+        return get_shard_of_pubkey(address.get_public_key())
```
Comment (ssd04) [inline multiversx_sdk_core/address.py:103]: in the specs it's specified that number of shards will be passes via the constructor, should it be done here?

File: multiversx_sdk_core/proto/transaction_serializer.py [hunk 3]
```diff
@@ -23,28 +23,20 @@ class ITransaction(Protocol):
...
-        ...
-
-    def bech32_to_pubkey(self, value: str) -> bytes:
-        ...
-
-
 class ProtoSerializer:
-    def __init__(self, converter: IAddressConverter) -> None:
-        self.address_converter = converter
+    def __init__(self) -> None:
```
Comment (andreibancioiu) [inline multiversx_sdk_core/proto/transaction_serializer.py:27]: Nice simplification!

**Review by andreibancioiu — COMMENTED** [review]
:+1: 

Only the comments from Darius.

File: multiversx_sdk_core/address.py (hunk 2 above)
Comment (popenta) [reply multiversx_sdk_core/address.py:103]: Passed it in via the constructor and is now passed to the `get_shard_of_pubkey()` function.

File: multiversx_sdk_core/address.py (hunk 1 above)
Comment (popenta) [reply multiversx_sdk_core/address.py:15]: renamed to `address`

**Review by andreibancioiu — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**